README.md

```diff
@@ -46,16 +52,57 @@ const client = SearchApplicationClient(applicationName, endpoint, apiKey, params
 ### Using the Client
 The Search Application Client provides a QueryBuilder class that allows you to build complex search queries:
 ```javascript
-const request = client();
-
-const results = await request
+const results = await client()
   .query('John')
-  .addFacetFilter('actors', 'Keanu Reeves')
-  .setSort({'year': 'desc'})
+  .setPage(1)
   .search()
 ```
 The search method returns a Promise that resolves with the search results.
 
+#### Use facets
+To use facets filtering, specify facets in the client initialization:
+```javascript
+const client = SearchApplicationClient('*search-application-name*', '*elasticsearch-endpoint*', '*apiKey*', {
+  facets: {
+    category: { type: 'terms', field: 'category.keyword', size: 10 },
+    price: { type: 'stats', field: 'price', size: 10, disjunctive: true },
+  },
+});
+
+const results = await client()
+  .addFacetFilter('category', 'electronics')
+  .addFacetFilter('price', { from: 100, to: 1000 })
+  .addFacetFilter('author', 'orwell') // ❌ Throw an error, because the facet is not specified during initialization
+  .setSort({'price': 'desc'})
+  .search()
+```
+
+#### Use sorting
+To use sorting, specify the field name and the sort order or pass _score to sort by relevance:
+```javascript
+const results = await client()
+  .setSort([{'publish_date': 'desc'}, "_score"])
+  .search()
+```
+
+#### Use pagination
+To use pagination, set the page number and the page size:
+```javascript
+const results = await client()
+  .setPage(2) // For the second page
```

Comment:
how does setPage know what is the second page if setPageSize hasn't been called? is there a default pageSize in the client?

Reply:
In query dsl size is 10 by default so I used that value, I can rollback a throw error when page size not set

Reply:
yeh i think best to keep it simple: setSize, setFrom - another challenge is the default parameter value if changed in the template wont impact the client computation.

Reply:
Ok I'll delete it then